obspy/core/util/testing.py

```diff
@@ -1050,5 +1050,167 @@ def setup_context_testcase(test_case, cm):
+        >>> # Unless, of course, there is a large change.
+        >>> st1[0].data *= 10
+        >>> assert not streams_almost_equal(st1, st2)
+    """
+    from obspy.core.stream import Stream
+    # Return False if both objects are not streams or not the same length.
+    are_streams = isinstance(st1, Stream) and isinstance(st2, Stream)
+    if not are_streams or not len(st1) == len(st2):
+        return False
+    # Kwargs to pass trace_almost_equal.
+    tr_kwargs = dict(default_stats=default_stats, rtol=rtol, atol=atol,
+                     equal_nan=equal_nan)
+    # Ensure the streams are sorted (as done with the __equal__ method)
+    st1_sorted = st1.select()
+    st1_sorted.sort()
+    st2_sorted = st2.select()
+    st2_sorted.sort()
+    # Iterate over sorted trace pairs and determine if they are almost equal.
+    for tr1, tr2 in zip(st1_sorted, st2_sorted):
+        if not traces_almost_equal(tr1, tr2, **tr_kwargs):
+            return False  # If any are not almost equal return None.
+    return True
+
+
+def traces_almost_equal(tr1, tr2, default_stats=True, rtol=1e-05, atol=1e-08,
+                        equal_nan=True):
+    """
+    Return True if the two traces are almost equal.
+
+    :param tr1: The first :class:`~obspy.core.trace.Trace` object.
+    :param tr2: The second :class:`~obspy.core.trace.Trace` object.
+    :param default_stats:
+        If True only compare the default stats on the traces, such as seed
+        identification codes, start/end times, sampling_rates, etc. If
+        False also compare extra stats attributes such as processing and
+        format specific information.
+    :param rtol: The relative tolerance parameter passed to
+        :func:`~numpy.allclose` for comparing time series.
+    :param atol: The absolute tolerance parameter passed to
+        :func:`~numpy.allclose` for comparing time series.
+    :param equal_nan:
+        If ``True`` NaNs are evaluated equal when comparing the time
+        series.
+    :return: bool
+    """
+    from obspy.core.trace import Trace
+    # If other isnt  a trace, or data is not the same len return False.
+    if not isinstance(tr2, Trace) or len(tr1.data) != len(tr2.data):
+        return False
+    # First compare trace stats objects.
+    stats1 = _make_stats_dict(tr1, default_stats)
+    stats2 = _make_stats_dict(tr2, default_stats)
+    if stats1 != stats2:
+        return False
+    # Then  compare the array values
+    try:  # Use equal_nan if available
+        all_close = np.allclose(tr1.data, tr2.data, rtol=rtol,
+                                atol=atol, equal_nan=equal_nan)
+    except TypeError:
+        # This happens on very old versions of numpy (< 1.7) which do not
+        # have isclose.
+        all_close = _all_close(tr1.data, tr2.data, rtol=rtol,
+                               atol=atol, equal_nan=equal_nan)
+    return all_close
+
+
+def _all_close(ar1, ar2, rtol, atol, equal_nan):
```

Comment:
Also move this to https://github.com/obspy/obspy/blob/master/obspy/core/compatibility.py.